### ISSUE TITLE
Unspecified "error" event

### ISSUE THREAD

**Issue by najawawe** [cazala/coin-hive-stratum#85]
Hello, thanks for your job with this proxy!

Regularly catching this exception leading to fall:
```
pool connection error (e73a9a71-fb7b-4cbe-9ac5-831f46d1a873): {"code":-1,"message":"Unauthenticated"}
miner disconnected (e73a9a71-fb7b-4cbe-9ac5-831f46d1a873)
events.js:165
      throw err;
      ^

Error: Uncaught, unspecified "error" event. ([object Object])
    at Miner.emit (events.js:163:17)
    at Miner.handleError (/root/.nvm/versions/node/v6.9.1/lib/node_modules/coin-hive-stratum/build/Miner.js:234:14)
    at emitOne (events.js:96:13)
    at Connection.emit (events.js:188:7)
    at Connection.receive (/root/.nvm/versions/node/v6.9.1/lib/node_modules/coin-hive-stratum/build/Connection.js:185:30)
    at Socket.<anonymous> (/root/.nvm/versions/node/v6.9.1/lib/node_modules/coin-hive-stratum/build/Connection.js:101:23)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
```
Can this be related to this issue?
https://stackoverflow.com/questions/45414114/error-uncaught-unspecified-error-event-object-object
Tried changing code (build/Miner.js +234) to this:
```
        this.sendToMiner({
            type: "error-data",
            params: error
        });
        this.emit("error-data", {
            id: this.id,
            login: this.login,
            error: error
        });
```
But yet had no chance of testing it :)

**Comment by cazala** [comment]
The emitted `error` messages are being handled by the proxy, the listeners are removed only when the miner disconnects. I think this is due to some async issue there the miner's errorHandler is executed after it has disconnected, so there's no listener for the miner's error event.

I added an `if` to [prevent the error from being emitted](https://github.com/cazala/coin-hive-stratum/commit/82a6d59015121d5ffca8b7ed06f09ea43733b91e#diff-c212081e48eee6e3704e2ce3ff96a0b0R209) if the miner is not online.

I also added just in case a [listener at the process level](https://github.com/cazala/coin-hive-stratum/commit/82a6d59015121d5ffca8b7ed06f09ea43733b91e#diff-f41e9d04a45c83f3b6f6e630f10117feR4) for all the unhandled exceptions. That should prevent something like this (or any other kind of exception) to stop the proxy.

I just released this as `v2.5.9`.
